src/Fixer/ConstantNotation/NativeConstantInvocationFixer.php

```diff
@@ -0,0 +1,234 @@
+                    ]
+                ),
+                new CodeSample(
+                    '<?php var_dump(PHP_VERSION, M_PI, MY_CUSTOM_PI);'.PHP_EOL,
+                    [
+                        'exclude' => [
+                            'M_PI',
+                        ],
+                    ]
+                ),
+            ],
+            null,
+            'Risky when any of the constants are namespaced or overridden.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $uniqueConfiguredExclude = \array_unique($this->configuration['exclude']);
+
+        // Case sensitive constants handling
+        $constantsToEscape = \array_values($this->configuration['include']);
+        if (true === $this->configuration['fix_built_in']) {
+            $getDefinedConstants = \get_defined_constants(true);
+            unset($getDefinedConstants['user']);
+            foreach ($getDefinedConstants as $constants) {
+                $constantsToEscape = \array_merge($constantsToEscape, \array_keys($constants));
+            }
+        }
+        $constantsToEscape = \array_diff(
+            \array_unique($constantsToEscape),
+            $uniqueConfiguredExclude
+        );
+
+        // Case insensitive constants handling
+        static $caseInsensitiveConstants = ['null', 'false', 'true'];
+        $caseInsensitiveConstantsToEscape = [];
+        foreach ($constantsToEscape as $constantIndex => $constant) {
+            $loweredConstant = \strtolower($constant);
+            if (\in_array($loweredConstant, $caseInsensitiveConstants, true)) {
+                $caseInsensitiveConstantsToEscape[] = $loweredConstant;
+                unset($constantsToEscape[$constantIndex]);
+            }
+        }
+
+        $caseInsensitiveConstantsToEscape = \array_diff(
+            \array_unique($caseInsensitiveConstantsToEscape),
+            \array_map('strtolower', $uniqueConfiguredExclude)
+        );
+
+        // Store the cache
+        $this->constantsToEscape = \array_fill_keys($constantsToEscape, true);
+        \ksort($this->constantsToEscape);
+
+        $this->caseInsensitiveConstantsToEscape = \array_fill_keys($caseInsensitiveConstantsToEscape, true);
+        \ksort($this->caseInsensitiveConstantsToEscape);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $useDeclarations = (new NamespaceUsesAnalyzer())->getDeclarationsFromTokens($tokens);
+        $useConstantDeclarations = [];
+        foreach ($useDeclarations as $use) {
+            if ($use->isConstant()) {
```

Comment:
@keradus in order to have this clean API, I extended the `NamespaceUsesAnalyzer` behaviour, which is backward compatible